### DISCUSSION TITLE
OCPP V1.6 SetChargingProfileRequest not valid

### DISCUSSION THREAD

**Discussion by ezolti** [ChargeTimeEU/Java-OCA-OCPP#332] [Q&A] [answered]
I have a charging center which I am trying to send to the charging point a profile.
This is the code
ChargingSchedulePeriod cp = new ChargingSchedulePeriod(1,Double.parseDouble(_settings.get("max.amper"))) ;
cp.setNumberPhases(1);
ChargingSchedulePeriod[] cpp = new ChargingSchedulePeriod[1];
cpp[0]=cp;
ChargingProfile chargingProfiles = new ChargingProfile(1,1,ChargingProfilePurposeType.ChargePointMaxProfile,
ChargingProfileKindType.Absolute,new ChargingSchedule(ChargingRateUnitType.A,cpp));
chargingProfiles.setTransactionId(_transactionid.get());

I can see in the log that the request is not valid.
SetChargingProfileRequest{connectorId=1, csChargingProfiles=ChargingProfile{chargingProfileId=1, transactionId=3, stackLevel=1, chargingProfilePurpose=ChargePointMaxProfile, chargingProfileKind=Absolute, recurrencyKind=null, validFrom="", validTo="", chargingSchedule=ChargingSchedule{duration=null, startSchedule="", chargingRateUnit=A, chargingSchedulePeriod.length=1, minChargingRate=null, isValid=true}, isValid=false}, isValid=false}

What is wrong?
@TVolden can you please help or point to the person who can?

**Comment by ezolti** [comment]
@TVolden can you please help or point to the person who can?

**Reply by Basler182** [reply, marked as answer]
The SetChargingProfileRequest contains a validate() method that you can look into for verification, of what is wrong in your request.

**Reply by ezolti** [reply]
thanks @Basler182 
I change to ChargingProfilePurposeType.TxProfile
It is working now
